extensions/vscode/src/templates/bloc.template.ts

```diff
@@ -57,3 +66,30 @@ class ${pascalCaseBlocName}Bloc extends Bloc<${blocEvent}, ${blocState}> {
 }
 `;
 }
+
+export function getFreezedBlocTemplate(blocName: string) {
+  const pascalCaseBlocName = changeCase.pascalCase(blocName.toLowerCase());
+  const snakeCaseBlocName = changeCase.snakeCase(blocName.toLowerCase());
+  const blocState = `${pascalCaseBlocName}State`;
+  const blocEvent = `${pascalCaseBlocName}Event`;
+  return `import 'dart:async';
+
+import 'package:bloc/bloc.dart';
+import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:meta/meta.dart';
+
+part '${snakeCaseBlocName}_event.dart';
+part '${snakeCaseBlocName}_state.dart';
+part '${snakeCaseBlocName}_bloc.freezed.dart';
+
+class ${pascalCaseBlocName}Bloc extends Bloc<${blocEvent}, ${blocState}> {
+  ${pascalCaseBlocName}Bloc() : super(_Initial());
+  @override
+  Stream<${blocState}> mapEventToState(
+    ${blocEvent} event,
+  ) async* {
+    // TODO: implement mapEventToState
+  }
```

Comment:
Minor improvement may be to actually implement an example event here, since we are creating an event in the event file. We can do this:
**Written without IDE, please verify functionality/syntax**
```suggestion
  @override
  Stream<${blocState}> mapEventToState(
    ${blocEvent} event,
  ) async* {
    yield* event.when(
      started: _mapStartedToState,
    );
  }
  
  Stream<${blocState> _mapStartedToState() async* {
    yield ${blocState}.loadInProgress();
    await Future.delayed(const Duration(seconds : 5));
    yield ${blocState}.loadSuccess();
  }
```

Reply:
I'm not sure about if we need to add the started event into bloc creation, because most blocs doesn't needs a `started` event. see [flutter examples](https://bloclibrary.dev/#/flutterfirebaselogintutorial?id=register-events) 
The intention of `event1` in the original event file is to allow the build_runner run successfully.

Reply:
Ah yeah. Fine by me, was just a thought 😄 
We also do have `feventToState` to create the method with 👍